integ-test/src/test/java/org/opensearch/sql/ppl/DedupCommandIT.java

```diff
@@ -62,4 +74,34 @@ public void testKeepEmptyDedup() throws IOException {
         rows("Virginia", null),
         rows("Dillard", 48086));
   }
+
+  private List<Object[]> extractActualRows(JSONObject result) {
+    JSONArray dataRows = result.getJSONArray("datarows");
+    List<Object[]> actualRows = new ArrayList<>();
+    for (int i = 0; i < dataRows.length(); i++) {
+      JSONArray row = dataRows.getJSONArray(i);
+      actualRows.add(new Object[] {row.get(0)});
+    }
+    return actualRows;
+  }
+
+  // Verify consecutive deduplication and create the expected rows
+  private List<Object[]> getExpectedDedupRows(List<Object[]> actualRows) {
```

Comment:
As I mentioned in the previous comment, we want to separate the logic to verify deduplication. (each method should have single responsibility for better maintainability)